client/src/main/scala/com.crobox.clickhouse/stream/ClickhouseSink.scala

```diff
@@ -11,15 +11,30 @@ import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
 
 case class ClickhouseIndexingException(msg: String, cause: Throwable, payload: Seq[String], table: String)
-    extends RuntimeException(msg, cause)
-case class Insert(table: String, jsonRow: String)
+  extends RuntimeException(msg, cause)
+
+sealed trait TableOperation {
+  def table: String
+
+  def jsonRow: String
+
+  def cluster: Option[String]
+}
+
+case class Insert(table: String, jsonRow: String) extends TableOperation {
+  override def cluster: Option[String] = None
+}
+
+case class Optimize(table: String, cluster: Option[String]) extends TableOperation {
+  override def jsonRow: String = ""
+}
 
 object ClickhouseSink extends LazyLogging {
 
-  def insertSink(config: Config, client: ClickhouseClient, indexerName: Option[String] = None)(
-      implicit ec: ExecutionContext,
-      settings: QuerySettings = QuerySettings()
-  ): Sink[Insert, Future[Done]] = {
+  def operationSink(config: Config, client: ClickhouseClient, indexerName: Option[String] = None)(
```

Comment:
In order to maintain backwards compatibility, please still maintain the 'insertSink' method but mark it as deprecated.
In the deprecation warning, please clearly state to use the 'operationSink', which should be renamed to 'toSink'